src/main/java/org/skife/jdbi/v2/BasicHandle.java

```diff
@@ -487,4 +490,20 @@ public void registerContainerFactory(ContainerFactory<?> factory)
     {
         this.containerFactoryRegistry.register(factory);
     }
+
+    @Override
+    public void setSqlObjectContext(Class<?> sqlObjectType, Method sqlObjectMethod) {
```

Comment:
Current work in progress here: https://github.com/jdbi/jdbi/commit/9e7aa0b5ff2598755fdaba47d6578dc26a574686

So there's some existing tests that expect the same handle to come back in certain circumstances--putting a wrapper around every handle sort of undermines that assumption.

At this point I think I'd rather stay with the approach on this PR.